### PR TITLE
Use validation library to validate closure and descriptive metadata

### PR DESCRIPTION
**Pull request by vimleshtna**

### CODE REVIEW

File: app/controllers/AddAdditionalMetadataController.scala [hunk 1]
```diff
@@ -129,6 +129,33 @@ class AddAdditionalMetadataController @Inject() (
     consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, Some(metadataType), Some(fileIds), additionalProperties)
   }
 
+  private def getMetadataValidation(consignmentId: UUID)(implicit request: Request[AnyContent]): Future[MetadataValidation] = {
```
Comment (TomJKing) [inline app/controllers/AddAdditionalMetadataController.scala:132]: Would it make more sense to have this method in the `MetadataValidationUtils.scala`?

**Review by TomJKing — COMMENTED** [review]
Looks good. Thanks. One minor comment. 

Also could you update the exploratory test Jira task with test cases to cover all the scenarios for adding form data: https://national-archives.atlassian.net/browse/TDR-3420

File: app/controllers/AddAdditionalMetadataController.scala (hunk 1 above)
Comment (vimleshtna) [reply app/controllers/AddAdditionalMetadataController.scala:132]: I created this `MetadataValidationUtils.scala` to process data. I don't think it's a good idea to call services from the utility class.

Comment (TomJKing) [reply app/controllers/AddAdditionalMetadataController.scala:132]: Fair enough

File: test/controllers/util/FormFieldSpec.scala [hunk 2]
```diff
@@ -513,6 +562,28 @@ class FormFieldSpec extends AnyWordSpec with MockitoSugar with BeforeAndAfterEac
       })
     }
 
+    "updateError should set an error message as per the given error code" in {
```
Comment (TomJKing) [inline test/controllers/util/FormFieldSpec.scala:565]: Can't see a check for the max character limit error in this test?

**Review by TomJKing — COMMENTED** [review]
Looks good. One minor comment.

File: test/controllers/util/FormFieldSpec.scala (hunk 2 above)
Comment (vimleshtna) [reply test/controllers/util/FormFieldSpec.scala:565]: Actually, the date field will never have a max character limit error. It is only applicable for `TextArea` and `RadioButtonGroupField` (dependency).

Comment (TomJKing) [reply test/controllers/util/FormFieldSpec.scala:565]: Fair enough

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**